### ISSUE TITLE
GPU usage and CPU bottleneck -- pretraining

### ISSUE THREAD

**Issue by olivierr42** [FlagOpen/FlagEmbedding#208]
I am currently trying to pretrain-then-finetune bge-base-en. However, given the script examples provided in this repo, my GPU is capped at around 40-50% of processing power on an A10 (VRAM is fully loaded). I was wondering whether the pretraining was known to be particularly CPU-intensive, which would explain the bottleneck I currently experience.

Thank you.

**Comment by staoxiao** [comment]
Yes, the pre-training needs to generate different masked sequences for each token, so the data processing is costly.  You can use a larger `dataloader_num_workers` to increase the utility ration of GPUs.

**Comment by olivierr42** [comment]
> Yes, the pre-training needs to generate different masked sequences for each token, so the data processing in CPU is costly. You can use a larger `dataloader_num_workers` to increase the utility ration of GPUs.

Thank you for the tip! It works.
